api/v3/Mailing.php

```diff
@@ -613,19 +614,24 @@ function civicrm_api3_mailing_send_test($params) {
   if (!array_key_exists('test_group', $params) && !array_key_exists('test_email', $params)) {
     throw new API_Exception("Mandatory key(s) missing from params array: test_group and/or test_email field are required");
   }
-  civicrm_api3_verify_mandatory($params,
-    'CRM_Mailing_DAO_MailingJob',
-    array('mailing_id'),
-    FALSE
-  );
+  $testEmailParams = [
+    'mailing_id' => $params['mailing_id'],
+    'is_test' => 1,
+    'status' => 'Scheduled',
+    'scheduled_date' => CRM_Utils_Date::processDate(date('Y-m-d'), date('H:i:s')),
+  ];
+  // It's unclear when / if there is a case for passing these through but defining this whitelist
+  // emulates prior behaviour, while preventing id being incorrectly passed.
+  $whitelist = ['start_date', 'end_date', 'parent_id', 'job_type', 'job_offset', 'job_limit'];
+  foreach ($whitelist as $field) {
+    if (isset($params[$field])) {
+      $testEmailParams[$field] = $params[$field];
+    }
+  }
 
-  $testEmailParams = _civicrm_api3_generic_replace_base_params($params);
```

Comment:
@eileenmcnaughton is this where the transposition was going on?

Reply:
@eileenmcnaughton do you think it might be worth adding id to the exclusion on the replace_base_params function instead?